hometask9/src/graph.ts

```diff
@@ -0,0 +1,99 @@
+import { iNode, iEdge } from "./interfaces";
+
+export class Graph {
+  list: iNode = {};
+
+  constructor() {}
+
+  addNode(node: string) {
+    this.list[node] = {};
+  }
+
+  removeNode(node: string) {
+    for (let listNode in this.list) {
+      this.removeEdge(node, listNode, true);
+    }
+    delete this.list[node];
+  }
+
+  addEdge(
+    fromNode: string,
+    toNode: string,
+    weight: number,
+    bothWays: boolean = false
+  ) {
+    if (this.list[fromNode] && this.list[toNode]) {
+      this.list[fromNode][toNode] = weight;
```

Comment:
Good! Very interesting representation on adjacency matrix. I like it much more than 2 dimensional array